File: swift/enumTest.swift

enum Gender{
    case M
    case F
}

func test(){
    let a = Gender.M
    print(a)
}

func test2(){
    let a = Gender.M
    switch a {
        case .M: print("male")
        case .F: print("female")
    }
}

test()

File: rust/test2/src/enumTest.rs

enum Sex{
	F(String),
	M{s:String},
	X
}

pub fn test(){
	let a = [
				Sex::F(String::from("female")), 
				Sex::M{s: String::from("male")}, 
				Sex::X
			];
	for k in a.iter(){
		match k {
			Sex::F(s) => println!("{}", s),
			Sex::M{s} => println!("{}", s),
			Sex::X => println!("{}", "orange")
		}
	}
}